### ISSUE TITLE
Preparing a statement with syntax error results in vtgate dropping connection

### ISSUE THREAD

**Issue by aquarapid** [vitessio/vitess#5734]
Scenario:

 * Cluster with unsharded keyspace and vschema:
```
{
  "tables": {
    "schema_migrations": {}
  }
}
```

 * SQL schema:
```
CREATE TABLE `schema_migrations` (`version` bigint(20) NOT NULL, `inserted_at` datetime DEFAULT NULL, PRIMARY KEY (`version`)) ENGINE=InnoDB DEFAULT CHARSET=utf8;
```

  * Run a program that prepares a statement with a syntax error, e.g. (This uses the official MySQL python connector):
```
import mysql.connector
from mysql.connector.cursor import MySQLCursorPrepared

cnx = mysql.connector.Connect(host='localhost', port=15306,
        database='commerce', user='root', charset='utf8', get_warnings=True)

curprep = cnx.cursor(cursor_class=MySQLCursorPrepared)
cur = cnx.cursor()

prepstmt = "delete s0 from schema_migrations as s0 where (s0.version = convert(?, unsigned)) limit ?"

curprep.execute(prepstmt)
cnx.commit()
```

  * Running this will fail with a lost connection:
```
$ python3 ~/prep.py 
Traceback (most recent call last):
  File "/home/jacques/prep.py", line 12, in <module>
    curprep.execute(prepstmt)
  File "/home/jacques/.local/lib/python3.7/site-packages/mysql/connector/cursor.py", line 1140, in execute
    self._prepared = self._connection.cmd_stmt_prepare(operation)
  File "/home/jacques/.local/lib/python3.7/site-packages/mysql/connector/connection.py", line 940, in cmd_stmt_prepare
    packet = self._send_cmd(ServerCmd.STMT_PREPARE, statement)
  File "/home/jacques/.local/lib/python3.7/site-packages/mysql/connector/connection.py", line 267, in _send_cmd
    return self._socket.recv()
  File "/home/jacques/.local/lib/python3.7/site-packages/mysql/connector/network.py", line 243, in recv_plain
    raise errors.InterfaceError(errno=2013)
mysql.connector.errors.InterfaceError: 2013: Lost connection to MySQL server during query
```

  * No errors in the vtgate info/error logs, and the prepared statement does not show up in the vtgate query logs.
  * Inspecting a tcpdump of the connection, vtgate seems to shut down the connection after the client sends the COM_PREPARE staetment (see attached image)

Looking through the code, it seems that we return a bare error at https://github.com/vitessio/vitess/blob/f7205b8a67a60fd493ba67611e02e23d8ef70bd2/go/mysql/conn.go#L856 after parsing the prepared statement, instead of forming and writing a MySQL error packet to the client.

I'll generate a PR to fix.

**Comment by aquarapid** [comment]
The wire trace:
![](file:///home/jacques/Pictures/prepare.png)

**Comment by aquarapid** [comment]
After this PR the script will return a reasonable error:
```
Traceback (most recent call last):
  File "/home/jacques/prep.py", line 12, in <module>
    curprep.execute(prepstmt)
  File "/home/jacques/.local/lib/python3.7/site-packages/mysql/connector/cursor.py", line 1140, in execute
    self._prepared = self._connection.cmd_stmt_prepare(operation)
  File "/home/jacques/.local/lib/python3.7/site-packages/mysql/connector/connection.py", line 941, in cmd_stmt_prepare
    result = self._handle_binary_ok(packet)
  File "/home/jacques/.local/lib/python3.7/site-packages/mysql/connector/connection.py", line 892, in _handle_binary_ok
    raise errors.get_exception(packet)
mysql.connector.errors.DatabaseError: 1105 (HY000): unknown error: syntax error at position 87 near 'limit'
```

to the vtgate error we output:

```
E0117 17:06:50.784081 1228504 conn.go:856] Conn client 1 (127.0.0.1:51652): Error parsing prepared statement: syntax error at position 87 near 'limit'
```

and we log the query to the vtgate query log (that was previously not happening):
```
Prepare 127.0.0.1:51982 root    'userData1'     'root'  2020-01-17 17:15:13.960703      2020-01-17 17:15:13.960707      0.000004        0.000000        0.000000       0.000000        DELETE  "delete s0 from schema_migrations as s0 where (s0.version = convert(?, unsigned)) limit ?"      map[]   0       0       "
```

**Comment by aquarapid** [comment]
Was fixed by #5735
